corehq/warehouse/management/commands/create_batch.py

```diff
@@ -1,26 +1,32 @@
 from __future__ import absolute_import
-from datetime import date
+from datetime import datetime
 from django.core.management import BaseCommand
 from corehq.warehouse.models import Batch
 
 
-USAGE = """Usage: ./manage.py create_batch
+USAGE = """Usage: ./manage.py create_batch <dag_slug>
 """
 
 
 class Command(BaseCommand):
     """
-    Example: ./manage.py create_batch
+    Example: ./manage.py create_batch app_status
     """
     help = USAGE
 
-    def handle(self, **options):
-        last_batch = Batch.objects.filter(completed_on__isnull=False).order_by('-end_datetime').first()
-        start = last_batch.end_datetime
-        end = date.today()
+    def add_arguments(self, parser):
+        parser.add_argument('dag_slug')
+
+    def handle(self, dag_slug, **options):
+        last_batch = Batch.objects.filter(dag_slug=dag_slug, completed_on__isnull=False).order_by('-end_datetime').first()
+        start = last_batch.end_datetime if last_batch else '2000-01-01'
+        end = datetime.utcnow()
```

Comment:
as i mentioned in [my comment](https://github.com/dimagi/commcare-hq/pull/19054#issuecomment-356097907) we're going to need to inspect if this dim or fact has dependencies and if so use the oldest end_datetime from all dependencies as the end time.

each dim and fact class should have the dependencies listed in the code

Reply:
I think we should try to push as much of the dependency logic to airflow as possible. If we make the batches self contained (one per dim for example), we can have airflow only run each batch when those dependencies have been fulfilled. Then we would not need to track dependencies in both airflow and our own code

Reply:
i think the only requirement here is to set the end datetime based on the oldest of the dependencies' batches end times because if you're loading a fact which links to data from a few different dims, then you need to make sure all your pulled fact data will have corresponding dim entries so that the foreign key references won't be blank. whether we do that here or within airflow doesn't really matter, but i don't think we can avoid doing this.